CONTRIBUTING.md

```diff
@@ -105,10 +105,14 @@ removed from the header in rpc responses as well.
 
 ## Branching Model and Release
 
-All repos should adhere to the branching model: http://nvie.com/posts/a-successful-git-branching-model/.
+We follow a variant of [git flow](http://nvie.com/posts/a-successful-git-branching-model/).
 This means that all pull-requests should be made against develop. Any merge to
 master constitutes a tagged release.
 
+Note all pull requests should be squash merged except for merging to master and
+merging master back to develop. This keeps the commit history clean and makes it
+easy to reference the pull request where a change was introduced.
```

Comment:
What if we want to include a PR that made it into develop *after* the release branch has been created? Squashing those changes will create another commit for the same changes. For example see: https://github.com/tendermint/tendermint/pull/3491 
If we squash merge this, it will create a commit that will make it to the release branch and then to master. The same changes exist under develop with a different commit.

Reply:
Can we enforce this through branch protection in github settings?

Reply:
Merging develop to release branch should be done with a normal merge, not squash. You're correct here.

Ideally we minimize this. In the ideal case, the release branch is checked out directly from develop and merged to master with no more changes. 

I suppose this means we should continue to do all review and finalization on develop before we make the release branch, so that ideally there's no PRs against only the release branch. 

But that would cause develop to freeze while we're prepping the release, which isn't ideal, so there's some tension here.

I don't think it's a big deal to have some merges from develop into release. But we can also just retarget PRs to the release branch to minimize the number of times we have new stuff on develop that we want in the release after we've already made the release branch ...

Reply:
> Can we enforce this through branch protection in github settings?

Which part?

I don't think we can enforce the squash merge rules :(